provider/scaleway/scaleway.go

```diff
@@ -0,0 +1,337 @@
+			continue
+		}
+		recordsToDelete[zone] = append(recordsToDelete[zone], endpointToScalewayRecordsChangeDelete(zone, c)...)
+	}
+
+	for _, c := range changes.Create {
+		zone, _ := zoneNameMapper.FindZone(c.DNSName)
+		if zone == "" {
+			log.Infof("Ignore record %s since it's not handled by ExternalDNS", c.DNSName)
+			continue
+		}
+		recordsToAdd[zone].Records = append(recordsToAdd[zone].Records, endpointToScalewayRecords(zone, c)...)
+	}
+	for _, c := range changes.UpdateNew {
+		zone, _ := zoneNameMapper.FindZone(c.DNSName)
+		if zone == "" {
+			log.Infof("Ignore record %s since it's not handled by ExternalDNS", c.DNSName)
+			continue
+		}
+		recordsToAdd[zone].Records = append(recordsToAdd[zone].Records, endpointToScalewayRecords(zone, c)...)
+	}
+
+	for _, zone := range dnsZones {
+		zoneName := getCompleteZoneName(zone)
+		req := &domain.UpdateDNSZoneRecordsRequest{
+			DNSZone: zoneName,
+			Changes: recordsToDelete[zoneName],
+		}
+		req.Changes = append(req.Changes, &domain.RecordChange{
+			Add: recordsToAdd[zoneName],
+		})
+		returnedRequests = append(returnedRequests, req)
+	}
+
+	return returnedRequests, nil
+}
+
+func getCompleteZoneName(zone *domain.DNSZone) string {
+	subdomain := zone.Subdomain + "."
+	if zone.Subdomain == "" {
+		subdomain = ""
+	}
+	return subdomain + zone.Domain
+}
+
+func endpointToScalewayRecords(zoneName string, ep *endpoint.Endpoint) []*domain.Record {
+	// no annotation results in a TTL of 0, default to 300 for consistency with other providers
+	var ttl = scalewyRecordTTL
+	if ep.RecordTTL.IsConfigured() {
+		ttl = uint32(ep.RecordTTL)
+	}
+	var priority = scalewayDefaultPriority
+	if prop, ok := ep.GetProviderSpecificProperty(scalewayPriorityKey); ok {
+		prio, err := strconv.ParseUint(prop.Value, 10, 64)
+		if err != nil {
+			log.Errorf("Failed parsing value of %s: %s: %v; using priority of %d", scalewayPriorityKey, prop.Value, err, scalewayDefaultPriority)
+		} else {
+			priority = uint32(prio)
+		}
+	}
+
+	records := []*domain.Record{}
+
+	for _, target := range ep.Targets {
+		records = append(records, &domain.Record{
+			Data:     target,
+			Name:     strings.Trim(strings.TrimSuffix(ep.DNSName, zoneName), ". "),
+			Priority: priority,
+			TTL:      ttl,
+			Type:     domain.RecordType(ep.RecordType),
+		})
+	}
+
+	return records
+}
+
+func endpointToScalewayRecordsChangeDelete(zoneName string, ep *endpoint.Endpoint) []*domain.RecordChange {
+	records := []*domain.RecordChange{}
+
+	for _, target := range ep.Targets {
+		records = append(records, &domain.RecordChange{
+			Delete: &domain.RecordChangeDelete{
+				Data: target,
+				Name: strings.Trim(strings.TrimSuffix(ep.DNSName, zoneName), ". "),
+				Type: domain.RecordType(ep.RecordType),
+			},
+		})
+	}
+
+	return records
+}
+
+func logChanges(req *domain.UpdateDNSZoneRecordsRequest) {
+	log.Infof("Updating zone %s", req.DNSZone)
+	if !log.IsLevelEnabled(log.InfoLevel) {
+		return
+	}
+	for _, change := range req.Changes {
```

Comment:
Don't want to log updates?

Reply:
Hum, addition and deletion are logged just below no?

Reply:
All good, my mistake 👍 .